src/pycparser/c_parser.py

```diff
@@ -123,7 +123,7 @@ def __init__(
         # saw: int name;
         # If 'name' is not a key in _scope_stack[n] then 'name' was not defined
         # in this scope at all.
-        self._scope_stack = [()]
+        self._scope_stack = {}
```

Comment:
I think its supposed to be 
`self._scope_stack = [{}]`
A list of an empty dict() element

In both places in this file

Reply:
It does show up eventually it seems
```
  File "/home/julian/Desktop/Datavenia/PipelineC/src/pycparser/c_parser.py", line 161, in _add_typedef_name
    if not self._scope_stack[-1].get(name, True):
KeyError: -1
```